scio-core/src/main/scala/com/spotify/scio/ScioContext.scala

```diff
@@ -136,19 +136,24 @@ private object RunnerContext {
     finalLocalArtifacts
   }
 
+  private val sanitizePath: String => String = _.replace("\\", "/")
+
+  private[scio] def isNonRepositoryEnvDir(s: String): Boolean = {
+    val sanitizedUserHome: String = sanitizePath(sys.props("user.home"))
+    s.matches(s"${sanitizedUserHome}/\\..+/.+") && !s.matches(
+      s"${sanitizedUserHome}/\\.(ivy2|m2|cache/coursier)/.+"
```

Comment:
We need to include `$HOME/Library/Caches/Coursier` as coursier as different directory on MacOs

Reply:
That won't be caught by the regex though since it's under `Library` and not a dot-directory, right?

Reply:
Sorry, you are right I misread this! all good!